x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/Mapper.java

```diff
@@ -113,6 +117,24 @@ public PhysicalPlan map(LogicalPlan p) {
             return map(ua, child);
         }
 
+        if (p instanceof BinaryPlan bp) {
+            var left = map(bp.left());
+            var right = map(bp.right());
+
+            if (left instanceof FragmentExec) {
+                if (right instanceof FragmentExec) {
+                    throw new EsqlIllegalArgumentException("can't plan binary [" + p.nodeName() + "]");
+                }
+                // in case of a fragment, push to it any current streaming operator
+                return new FragmentExec(p);
+            }
+            if (right instanceof FragmentExec) {
+                // in case of a fragment, push to it any current streaming operator
+                return new FragmentExec(p);
+            }
```

Comment:
Yeah, I mirrored the other code and it led to quite the abstraction layers. 

To be honest this is all stuff I'm a lot less familiar with.

Reply:
> we won't have general binary plans, because we cannot map them.

We don't have _yet_ binary plans, but this PR is the first step in the right direction. see https://github.com/elastic/elasticsearch/issues/99841